Source/WebKit/UIProcess/mac/WebViewImpl.mm

```diff
@@ -5188,7 +5188,10 @@ static BOOL shouldUseHighlightsForMarkedText(NSAttributedString *string)
 #if HAVE(INLINE_PREDICTIONS)
 bool WebViewImpl::allowsInlinePredictions() const
 {
-    return m_page->preferences().inlinePredictionsEnabled() && [NSSpellChecker respondsToSelector:@selector(isAutomaticInlineCompletionEnabled)] && NSSpellChecker.isAutomaticInlineCompletionEnabled;
+    if (!m_page->preferences().inlinePredictionsEnabled() && !m_page->preferences().inlinePredictionsInAllEditableElementsEnabled())
```

Comment:
Now that I'm reading this, I think we should (in a separate PR) look into getting rid of the original `InlinePredictionsEnabled` from UnifiedWebPreferences. Since that preference isn't read in WebCore, or in the WebProcess, we could simply read off the `WKWebViewConfiguration`, without a need for a preference associated with the API.

Note how iOS already does this by checking `self.webView.configuration.allowsInlinePredictions`. The only problem is that `WebViewImpl` doesn't have easy access to `WKWebViewConfiguration` right now, which is why I suggest a separate PR.

Reply:
> The only problem is that `WebViewImpl` doesn't have easy access to `WKWebViewConfiguration`

(Would it work to use `m_page->configuration()`, which returns the backing `API::Configuration`? Or alternately, `[m_view configuration]` should give the `WKWebViewConfiguration`, but at the cost of an extra copy of the configuration which is not ideal).